src/platforms/web/runtime/modules/events.js

```diff
@@ -3,13 +3,27 @@
 
 import { updateListeners } from 'core/vdom/helpers/index'
 
+const onceFlags = {}
+const randomHex = () => (Math.random() + '').substr(2).toString(16)
+
 function updateDOMListeners (oldVnode, vnode) {
   if (!oldVnode.data.on && !vnode.data.on) {
     return
   }
   const on = vnode.data.on || {}
   const oldOn = oldVnode.data.on || {}
-  const add = vnode.elm._v_add || (vnode.elm._v_add = (event, handler, capture) => {
+  const add = vnode.elm._v_add || (vnode.elm._v_add = (event, handler, capture, once) => {
+    if (once) {
+      const randomKey = randomHex() + new Date().valueOf().toString(16) + randomHex()
+      onceFlags[randomKey] = false
+      const oldHandler = handler
+      handler = () => {
+        if (!onceFlags[randomKey]) {
```

Comment:
That's a good question.

Actually, the pr is way not well-tested. I'm working on it.